opentech/static_src/src/app/src/components/GroupedListing.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ListingGroup from '@components/ListingGroup';
+import ListingItem from '@components/ListingItem';
+import LoadingPanel from '@components/LoadingPanel';
+
+import '@components/Listing/style.scss';
+
+export default class GroupedListing extends React.Component {
+    static propTypes = {
+        items: PropTypes.array,
+        activeItem: PropTypes.number,
+        isLoading: PropTypes.bool,
+        error: PropTypes.string,
+        groupBy: PropTypes.string,
+        order: PropTypes.arrayOf(PropTypes.string),
+        onItemSelection: PropTypes.func,
+    };
+
+    state = {
+        orderedItems: [],
+    };
+
+    componentDidMount() {
+        this.orderItems();
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        // Order items
+        if (this.props.items !== prevProps.items) {
+            this.orderItems();
+        }
+
+        const oldItem = prevProps.activeItem
+        const newItem = this.props.activeItem
+
+        // If we have never activated a submission, get the first item
+        if ( !newItem && !oldItem ) {
+            const firstGroup = this.state.orderedItems[0]
+            if ( firstGroup && firstGroup.items[0] ) {
+                this.setState({firstUpdate: false})
+                this.props.onItemSelection(firstGroup.items[0].id)
+            }
+        }
+    }
+
+    renderListItems() {
+        const { isLoading, error, items, onItemSelection, activeItem } = this.props;
+
+        if (isLoading) {
+            return (
```

Comment:
Can we not delegate this to Listing such that:

```
<GroupedListing>
    <Listing onError={}>
</GroupedListing>
```

This way grouped listing is all about grouping?